core-rust/state-manager/src/staging/result.rs

```diff
@@ -264,6 +245,35 @@ impl ProcessedCommitResult {
             .map(|next_epoch_result| NextEpoch::from(next_epoch_result.clone()))
     }
 
+    pub fn compute_substate_changes<S: SubstateDatabase, D: DatabaseKeyMapper>(
+        store: &S,
+        system_updates: &SystemUpdates,
```

Comment:
Ideally this could take an owned SystemUpdates so we can avoid unnecessary cloning?

Reply:
unfortunately, with the current shape of the callers, we wouldn't gain much - the entire `CommitResult` is needed to create a `LocalTransactionReceipt` (after `compute_substate_changes()`), so we wouldn't be able to move out the `system_updates` anyways.